### PR TITLE
create storage interfaces

### PR DESCRIPTION
**Pull request by cli1903**

### CODE REVIEW

File: frontend/src/main/java/com/google/tripmeout/frontend/PlaceVisitStorage.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package com.google.tripmeout.frontend;
```
Comment (KoopaKing) [inline frontend/src/main/java/com/google/tripmeout/frontend/PlaceVisitStorage.java]: can you put the storage stuff in its own package?

File: frontend/src/main/java/com/google/tripmeout/frontend/PlaceVisitStorage.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+
+import com.google.tripmeout.frontend.PlaceVisitModel;
+
+public interface PlaceVisitStorage {
+  public void addPlaceVisit(PlaceVisitModel placeVisit);
```
Comment (KoopaKing) [inline frontend/src/main/java/com/google/tripmeout/frontend/PlaceVisitStorage.java]: Can you add some comments about what the expected behaviors are for these methods?
- For `addPlaceVisit` what is the expected behavior of implementations when the place already exists? Does it replace the existing object? Does it throw an exception? Does it fail silently?
- For `removePlaceVisit`, `getPlaceVisit` and `changePlaceVisitStatus` what is the expected behavior of implementations when the place doesn't exist?

File: frontend/src/main/java/com/google/tripmeout/frontend/TripStorage.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+import com.google.tripmeout.frontend.TripModel;
+import com.google.tripmeout.frontend.PlaceVisitModel;
+import java.util.List;
+
+public interface TripStorage {
+  public void addTrip(TripModel trip);
+  public void removeTrip(String tripId);
+  public TripModel getTrip(String tripId);
+  public List<TripModel> getAllUserTrips(String userId);
+  public List<PlaceVisitModel> getTripPlaceVisits(String tripId);
```
Comment (KoopaKing) [inline frontend/src/main/java/com/google/tripmeout/frontend/TripStorage.java]: Should `getTripPlaceVisits` be in PlaceVisitStorage?

File: frontend/src/main/java/com/google/tripmeout/frontend/TripStorage.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package com.google.tripmeout.frontend;
+
+import com.google.tripmeout.frontend.TripModel;
+import com.google.tripmeout.frontend.PlaceVisitModel;
+import java.util.List;
+
+public interface TripStorage {
+  public void addTrip(TripModel trip);
```
Comment (KoopaKing) [inline frontend/src/main/java/com/google/tripmeout/frontend/TripStorage.java]: Same comments here as on PlaceVisitStorage

**Review by chuckluv — DISMISSED** [review]

File: frontend/src/main/java/com/google/tripmeout/frontend/storage/PlaceVisitStorage.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+public interface PlaceVisitStorage {
+  /**
+   * adds a PlaceVisitModel object to storage
+   * 
+   * @param placeVisit the PlaceVisitModel object to add to storage
+   *
+   * @throws a PlaceAlreadyExists exception if there is a PlaceVisitModel in 
+   *     storage with the same tripId and placeId as placeVisit
+   */
+  public void addPlaceVisit(PlaceVisitModel placeVisit);
```
Comment (KoopaKing) [inline frontend/src/main/java/com/google/tripmeout/frontend/storage/PlaceVisitStorage.java]: nit: Interfaces don't require public qualifiers for their method definitions. You still need the public identifier on the interface itself, but can remove the rest.

File: frontend/src/main/java/com/google/tripmeout/frontend/storage/TripStorage.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+   *     already in storage with the same tripId as the trip
+   */
+  public void addTrip(TripModel trip);
+
+  /**
+   * removes from storage the TripModel object whose tripId matches the given tripId
+   *
+   * @param tripId the id to match TripModel object's tripId field on
+   *
+   * @throws a TripModelNotFound exception if there is no TripModel object with
```
Comment (KoopaKing) [inline frontend/src/main/java/com/google/tripmeout/frontend/storage/TripStorage.java]: Should this exception just be `TripNotFound` for continuity with the places exceptions?

File: frontend/src/main/java/com/google/tripmeout/frontend/storage/TripStorage.java [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+   *     the given tripId in storage
+   */
+  public TripModel getTrip(String tripId);
+
+  /**
+   * returns all TripModel objects in storage whose userId matches the given userId
+   *
+   * @param userId the id to match TripModel object's userId field on
+   */
+  public List<TripModel> getAllUserTrips(String userId);
```
Comment (KoopaKing) [inline frontend/src/main/java/com/google/tripmeout/frontend/storage/TripStorage.java]: Don't need to change it, but this method has an interesting connotation about what your storage model looks like.

**Review by KoopaKing — DISMISSED** [review]
Please address the nits.

File: frontend/src/main/java/com/google/tripmeout/frontend/storage/TripStorage.java (hunk 7 above)
Comment (cli1903) [reply frontend/src/main/java/com/google/tripmeout/frontend/storage/TripStorage.java]: I'm not really sure what you mean by this. I was just imagining a list of TripModel objects for the in memory storage

**Review by KoopaKing — APPROVED** [review]

**Review by chuckluv — APPROVED** [review]

**Pull request merged**